### ISSUE TITLE
[maybe SteamVR Input] ViveCraft Linux with Valve Index controllers behaving weird

### ISSUE THREAD

**Issue by airzocker** [jrbudda/Vivecraft_114#47]
I'm on linux and using the valve index. I finally got vivecraft (at least 1.12.2+) running and loading into worlds, but the inventory is constantly opening and closing on it's own (like spamming the inventory button) except i'm holding down the inventory button. I also can't switch hotbar slots so i suspect the same thing is happening there and most annoyingly the controllers are vibrating all the time since game startup. It seems like these in- and outputs are somehow inverted, but i've checked the steamVR input settings for vivecraft and it isn't only the default configuration for the index controllers but they seem like they are as they should (steam games run absolutely fine through steamVR) so what can i do? I really want to play vivecraft ;D

If you need any logs or something i can provide that, but it's not crashing and i don't think it's outputting input information so i didn't include any logs...

thanks for your help ;D

**Comment by Techjar** [comment]
Something tells me Linux support for the Valve Index is not in great shape...

**Comment by airzocker** [comment]
Hmm, what side do you think has lacking linux support? Because, as i said, steam games run absolutely flawlessly and i'm pretty sure i could change the steamVR Input controls (for every game) and it wouldn't have a problem. So do you think it's something with vivecraft?

**Comment by TheTrueForce** [comment]
It'll be SteamVR on Linux being fucky. It's been that way basically from day one.

**Comment by airzocker** [comment]
Well, i'll try changing the buttons a bit today and we'll see...

**Comment by airzocker** [comment]
Well, i'll try changing the buttons a bit today and we'll see...

**Comment by airzocker** [comment]
so i've tried around with the input configuration and now it's running perfectly fine for me (i had to change some controls to avoid the broken ones), the only thing is that i can't change hotbar slots with the two buttons, but luckily you can select hotbar slots with your other hand ;D (and as i said, only 1.12.2 is running for me)
so let's hope steamvr fixes the linux input some time...

**Comment by Techjar** [comment]
Yeah everything works fine on Windows so it must be something wrong with SteamVR on Linux. Most SteamVR games on Steam are still using legacy input; very few have upgraded to the new SteamVR Input API, so I suspect that's what's broken on Linux. Last I heard legacy input works fine, so that explains why all the other stuff you've tried has no issues.

**Comment by airzocker** [comment]
i think that might be it, but 1.7.10 which is using legacy input isn't working for me (it freezes when loading a world...), maybe i'll look into that later ;D

**Comment by Techjar** [comment]
Try v8 of 1.12.2

**Comment by airzocker** [comment]
ok, downloading right now, i'll try it tomorrow.

**Comment by mnn** [comment]
> the inventory is constantly opening and closing on it's own (like spamming the inventory button) except when i'm holding down the inventory button ...
> and if i press down the trackads the game thinks they are not pressed and also the vibration of the pressed controller stops

Having exactly same issue.

> so i've tried around with the input configuration and now it's running perfectly fine for me (i had to change some controls to avoid the broken ones)

Hmm, will have to try more changes to controls then. Tried few popular presets and none worked.

MC 1.12.2
ViveCraft 10-r3.
Linux (Manjaro)
SteamVR 1.13.10 (beta branch behaves same)

**Comment by airzocker** [comment]
yeah, I have also given up...
